File: src/main/java/com/otaviolarrosa/identitymanager/commons/api/ApiResultBase.java
package com.otaviolarrosa.identitymanager.commons.api;

import java.util.ArrayList;
import java.util.List;

public class ApiResultBase {
    
	public ApiResultBase(){
        setErrors(new ArrayList<Error>());
    }
    
    private List<Error> errors;
    public List<Error> getErrors() {
        return errors;
    }

    public void setErrors(List<Error> errors) {
        this.errors = errors;
    }

    public boolean getIsValid() {
        return getErrors().isEmpty();
    }
    
    
    public void AddError(String errorMessage) {
    	this.errors.add(new Error(errorMessage));
    }
}

File: src/main/kotlin/com/otaviolarrosa/singleSignOn/api/commons/ApiResultBase.kt
package com.otaviolarrosa.singleSignOn.api.commons

open class ApiResultBase {
    val errors: MutableList<Error> = mutableListOf()
    val valid  : Boolean
        get() = this.errors.isEmpty()
    val invalid  : Boolean
        get() = this.errors.isNotEmpty()

    fun addError(errorMessage: String) {
        val error = Error(errorMessage)
        errors.addLast(error)
    }
}